Complexity_classes_def/P.thy

```diff
@@ -401,10 +401,11 @@ definition IMP_SAT :: "nat set" where
 "IMP_SAT == encode_sat ` {n. sat n}"
```

Comment:
I think we should mention a time and space bound on encode and decode (This has no effect whatsoever on our work). But so that our work really reflects the meaning of complexity classes, the encoding/decoding should be in polynomial time, otherwise, one could cherat and use the decoding to kind of solve the problem inside.Thus every decidable problem could be in P/NP that way.